posttroll/ns.py

```diff
@@ -119,19 +121,82 @@ def __init__(self, max_age=None, multicast_enabled=True, restrict_to_localhost=F
         from posttroll.backends.zmq.ns import ZMQNameServer
         self._ns = ZMQNameServer()
 
-    def run(self, *args):
+    def run(self, address_receiver=None, nameserver_address=None):
         """Run the listener and answer to requests."""
-        del args
-
-        arec = AddressReceiver(max_age=self._max_age,
-                               multicast_enabled=self._multicast_enabled,
-                               restrict_to_localhost=self._restrict_to_localhost)
-        arec.start()
+        if address_receiver is None:
+            address_receiver = AddressReceiver(max_age=self._max_age,
+                                               multicast_enabled=self._multicast_enabled,
+                                               restrict_to_localhost=self._restrict_to_localhost)
+            address_receiver.start()
         try:
-            return self._ns.run(arec)
+            return self._ns.run(address_receiver, nameserver_address)
         finally:
-            arec.stop()
+            with suppress(AttributeError):
+                address_receiver.stop()
 
     def stop(self):
         """Stop the nameserver."""
         return self._ns.stop()
+
+
+def main():
```

Comment:
Could the runnable script stay outside of the library? Or is there a benefit of moving it within?

Reply:
yes, we can then have it as a script in pyproject.toml if it’s defined in the library

Reply:
Right, so the scripts actually can't be outside of the package with pyproject.toml :see_no_evil: I've liked to have them outside so that they don't affect the test coverage :sweat_smile: